File: 이것이코딩테스트다/simulation/왕실의나이트.js
const fs = require('fs');
const input = fs.readFileSync('input.txt').toString().trim();

let col = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
let [cur_col, cur_row] = input.split('')
let answer = 0;

cur_row = parseInt(cur_row);
cur_col = col.indexOf(cur_col) + 1


let dx = [-1, 1, -2, 2, 2, -2, -1, 1]
let dy = [-2, 2, -1, 1, -1, 1, 2, -2]

for (let i = 0; i < 8; i++) {
  let nx = cur_col + dx[i]
  let ny = cur_row + dy[i]
  if (nx > 0 && nx <= 8 && ny > 0 && ny <= 8) {
    answer += 1
  }
}

console.log(answer)

File: 이것이코딩테스트다/simulation/왕실의나이트.py
start = input()
row = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
cur_col = int(row.index(start[0]) + 1)
cur_row = int(start[1])

print(cur_col, cur_row)

dx = [-1 ,1, -2 ,2, 2, -2, -1, 1]
dy = [-2, 2, -1, 1, -1, 1, 2, -2]

answer = 0

for i in range(8):
  nx = cur_col + dx[i]
  ny = cur_row + dy[i]
  if nx > 0 and nx <= 8 and ny > 0 and ny <= 8:
    answer += 1

print(answer)